### ISSUE TITLE
New tif files disappearing?

### ISSUE THREAD

**Issue by hyacin75** [VSF1/minolta2exif#5]
Thank you so much for picking this up and updating it!!  VERY hard to even find the old posts and links to the script any more, so I was so happy to find your repo!

I've got the script seemingly running as expected, it plugs along happily -

```
me@manager:~/film-scans$ ./dn7toexif.pl dn1-000@R_@F DN1-000*.TXT
Processing DN1-0001.TXT

ISO found 400
Modifying dn1-0001_01.tif into dn1-0001_01-exif.tif
Modifying dn1-0001_02.tif into dn1-0001_02-exif.tif
Modifying dn1-0001_03.tif into dn1-0001_03-exif.tif
Modifying dn1-0001_04.tif into dn1-0001_04-exif.tif
Modifying dn1-0001_05.tif into dn1-0001_05-exif.tif
^C
```

and it creates the -exif.tif files and the .xmp files as it's going -

```
me@DS220:~/film-scans$ ls
Developed-RAW         dn1-0001_07.tif  dn1-0001_12.tif  dn1-0001_17.tif  dn1-0001_22.tif  dn1-0001_27.tif  dn1-0001_32.tif  DN1-0001.TXT     dn1-0002_10.tif  dn1-0002_20.tif  dn1-0002_30.tif
dn1-0001_01.tif       dn1-0001_07.xmp  dn1-0001_12.xmp  dn1-0001_17.xmp  dn1-0001_22.xmp  dn1-0001_27.xmp  dn1-0001_32.xmp  dn1-0002_01.tif  dn1-0002_11.tif  dn1-0002_21.tif  dn1-0002_31.tif
dn1-0001_02.tif       dn1-0001_08.tif  dn1-0001_13.tif  dn1-0001_18.tif  dn1-0001_23.tif  dn1-0001_28.tif  dn1-0001_33.tif  dn1-0002_02.tif  dn1-0002_12.tif  dn1-0002_22.tif  dn1-0002_32.tif
dn1-0001_03.tif       dn1-0001_08.xmp  dn1-0001_13.xmp  dn1-0001_18.xmp  dn1-0001_23.xmp  dn1-0001_28.xmp  dn1-0001_33.xmp  dn1-0002_03.tif  dn1-0002_13.tif  dn1-0002_23.tif  dn1-0002_33.tif
dn1-0001_04-exif.tif  dn1-0001_09.tif  dn1-0001_14.tif  dn1-0001_19.tif  dn1-0001_24.tif  dn1-0001_29.tif  dn1-0001_34.tif  dn1-0002_04.tif  dn1-0002_14.tif  dn1-0002_24.tif  dn1-0002_34.tif
dn1-0001_04.tif       dn1-0001_09.xmp  dn1-0001_14.xmp  dn1-0001_19.xmp  dn1-0001_24.xmp  dn1-0001_29.xmp  dn1-0001_34.xmp  dn1-0002_05.tif  dn1-0002_15.tif  dn1-0002_25.tif  dn1-0002_35.tif
dn1-0001_05.tif       dn1-0001_10.tif  dn1-0001_15.tif  dn1-0001_20.tif  dn1-0001_25.tif  dn1-0001_30.tif  dn1-0001_35.tif  dn1-0002_06.tif  dn1-0002_16.tif  dn1-0002_26.tif  dn1-0002_36.tif
dn1-0001_05.xmp       dn1-0001_10.xmp  dn1-0001_15.xmp  dn1-0001_20.xmp  dn1-0001_25.xmp  dn1-0001_30.xmp  dn1-0001_35.xmp  dn1-0002_07.tif  dn1-0002_17.tif  dn1-0002_27.tif  DN1-0002.TXT
dn1-0001_06.tif       dn1-0001_11.tif  dn1-0001_16.tif  dn1-0001_21.tif  dn1-0001_26.tif  dn1-0001_31.tif  dn1-0001_36.tif  dn1-0002_08.tif  dn1-0002_18.tif  dn1-0002_28.tif  dn7toexif.pl
dn1-0001_06.xmp       dn1-0001_11.xmp  dn1-0001_16.xmp  dn1-0001_21.xmp  dn1-0001_26.xmp  dn1-0001_31.xmp  dn1-0001_36.xmp  dn1-0002_09.tif  dn1-0002_19.tif  dn1-0002_29.tif
me@DS220:~/film-scans$
```

but then the -exif.tif files disappear when it moves on to the next one!

```
me@DS220:~/film-scans$ ls
Developed-RAW         dn1-0001_07.xmp  dn1-0001_12.xmp  dn1-0001_17.xmp  dn1-0001_22.xmp  dn1-0001_27.xmp  dn1-0001_32.xmp  dn1-0002_01.tif  dn1-0002_11.tif  dn1-0002_21.tif  dn1-0002_31.tif
dn1-0001_01.tif       dn1-0001_08.tif  dn1-0001_13.tif  dn1-0001_18.tif  dn1-0001_23.tif  dn1-0001_28.tif  dn1-0001_33.tif  dn1-0002_02.tif  dn1-0002_12.tif  dn1-0002_22.tif  dn1-0002_32.tif
dn1-0001_02.tif       dn1-0001_08.xmp  dn1-0001_13.xmp  dn1-0001_18.xmp  dn1-0001_23.xmp  dn1-0001_28.xmp  dn1-0001_33.xmp  dn1-0002_03.tif  dn1-0002_13.tif  dn1-0002_23.tif  dn1-0002_33.tif
dn1-0001_03.tif       dn1-0001_09.tif  dn1-0001_14.tif  dn1-0001_19.tif  dn1-0001_24.tif  dn1-0001_29.tif  dn1-0001_34.tif  dn1-0002_04.tif  dn1-0002_14.tif  dn1-0002_24.tif  dn1-0002_34.tif
dn1-0001_04.tif       dn1-0001_09.xmp  dn1-0001_14.xmp  dn1-0001_19.xmp  dn1-0001_24.xmp  dn1-0001_29.xmp  dn1-0001_34.xmp  dn1-0002_05.tif  dn1-0002_15.tif  dn1-0002_25.tif  dn1-0002_35.tif
dn1-0001_05-exif.tif  dn1-0001_10.tif  dn1-0001_15.tif  dn1-0001_20.tif  dn1-0001_25.tif  dn1-0001_30.tif  dn1-0001_35.tif  dn1-0002_06.tif  dn1-0002_16.tif  dn1-0002_26.tif  dn1-0002_36.tif
dn1-0001_05.tif       dn1-0001_10.xmp  dn1-0001_15.xmp  dn1-0001_20.xmp  dn1-0001_25.xmp  dn1-0001_30.xmp  dn1-0001_35.xmp  dn1-0002_07.tif  dn1-0002_17.tif  dn1-0002_27.tif  DN1-0002.TXT
dn1-0001_06.tif       dn1-0001_11.tif  dn1-0001_16.tif  dn1-0001_21.tif  dn1-0001_26.tif  dn1-0001_31.tif  dn1-0001_36.tif  dn1-0002_08.tif  dn1-0002_18.tif  dn1-0002_28.tif  dn7toexif.pl
dn1-0001_06.xmp       dn1-0001_11.xmp  dn1-0001_16.xmp  dn1-0001_21.xmp  dn1-0001_26.xmp  dn1-0001_31.xmp  dn1-0001_36.xmp  dn1-0002_09.tif  dn1-0002_19.tif  dn1-0002_29.tif
dn1-0001_07.tif       dn1-0001_12.tif  dn1-0001_17.tif  dn1-0001_22.tif  dn1-0001_27.tif  dn1-0001_32.tif  DN1-0001.TXT     dn1-0002_10.tif  dn1-0002_20.tif  dn1-0002_30.tif
me@DS220:~/film-scans$
```

The XMP files remain though.  The originals don't have the EXIF data, so it's not like the new ones are being copied over the old ones or anything ... so I'm totally stumped as to what is going on here!  I tried commenting out the XMP writeinfo in case that was somehow trumping the previous operation, but it's still doing the same thing - after a full run, I have XMP for everything, my original .tif files, and that's it!

I don't think this is the expected behaviour, is it?  When I tried loading the entire directory up in Lightroom the XMP files weren't doing anything and all my .tif files still had no EXIF data, so I'm guessing and was under the impression that I am in fact supposed to end the process with my old files, new ones with EXIF data, plus XMP files.  Am I misunderstanding?

Thanks!!!

**Comment by hyacin75** [comment]
I just tried pulling and printing any errors and warnings from the operation but I don't think what it's saying is of note -

```
ISO found 400
Modifying dn1-0001_01.tif into dn1-0001_01-exif.tif
Errors: [minor] Maker notes could not be parsed
Use of uninitialized value $warningMessage in concatenation (.) or string at ./dn7toexif.pl line 285.
Warnings:

```

**Comment by hyacin75** [comment]
OK it looks like it actually is of note as I'm getting a return code 0 and it _is_ an error, not a warning -

```
rob@manager:~/film-scans$ ./dn7toexif.pl dn1-000@R_@F DN1-000*.TXT
Processing DN1-0001.TXT

ISO found 400
Modifying dn1-0001_01.tif into dn1-0001_01-exif.tif
Return code: 0
Errors: [minor] Maker notes could not be parsed
Use of uninitialized value $warningMessage in concatenation (.) or string at ./dn7toexif.pl line 286.
Warnings:

```

module docs say return code 0 is - `0 on file write error`

I'm finding a lot of things that say "use -m or -F" but that doesn't help when you're calling the WriteInfo function as opposed to using the cli tool 😞 ... I don't see anything in the WriteInfo docs that indicate how to do those same things ...

**Comment by hyacin75** [comment]
Alright, after much digging through the exiftool code I found the following to do the same as `-m` -

`$exifTool->Options(IgnoreMinorErrors => 1)`

I added that right before writing out and it seems to be plugging along fine now.

`-F` would have been preferred, but it takes additional arguments and thinking and I don't care enough to dive in that deeply.

I'll close this off now as it's 'fixed' enough for me.
